ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def order(request):
+    return render(request, 'ordered_product/new_order.html', {})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def search_product_by_product_name(request):
+    prod_name = str(request.GET['product_name'])
+    message = ""
+    try:
+        product = Product.objects.get(product_name=prod_name)
+        supplier_products = SupplierProduct.objects.filter(qr_code=product)
+    except ObjectDoesNotExist:
+        supplier_products = []
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def get_products_by_client_area(request):
+    message = ""
+    client = client_check(request.user)
+    supplier_products = []
+    try:
+        deliveries_in_client_location = DeliveryLocation.objects.filter(location=client.area)
+        for delivery in deliveries_in_client_location:
+            supplier_products += list(SupplierProduct.objects.filter(user_name=delivery.user_name))
+    except Exception:
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def get_products_by_supplier_user_name(request):
+    supplier_user_name = str(request.GET['supplier_user_name'])
+    message = ""
+    supplier_has_delivery_in_client_area = False
+    try:
+        user = User.objects.get(username=supplier_user_name)
+        supplier = Supplier.objects.get(supplier_account=user)
+        deliveries = DeliveryLocation.objects.filter(user_name=supplier)
+        for delivery in deliveries:
+            if delivery.location == client_check(request.user).area:
+                supplier_has_delivery_in_client_area = True
+        if not supplier_has_delivery_in_client_area:
+            raise ObjectDoesNotExist
+        supplier_products = SupplierProduct.objects.filter(user_name=supplier)
+    except ObjectDoesNotExist:
+        supplier_products = []
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def find_delivery(request, supplier_product_id):
+    try:
+        supplier_product = SupplierProduct.objects.get(supplier_product_id=int(supplier_product_id))
+    except ObjectDoesNotExist:
+        return redirect('/client/')
+    client = client_check(request.user)
+    deliveries = DeliveryLocation.objects.filter(user_name=supplier_product.user_name, location=client.area)
```

Comment:
You have a method for that:
```suggestion
    deliveries = DeliveryLocation.filter_by_supplier_and_location(supplier_product.user_name, client.area)
```